### PR TITLE
Search implemented in fragments.

### PR DESCRIPTION
**Pull request by m-murad**
Fixes issue #173 

@garg-anuj 
Sorry for being so late. 
To solve this I had to solve #186 

Please merge if I'm allowed to do that.

### CODE REVIEW

**Comment by pilhuhn** [comment]
@free4murad Can you add some screenshots please?

**Comment by m-murad** [comment]
@pilhuhn 
Sure 
Let me upload a video instead so that things are more clear.
https://drive.google.com/open?id=0ByRo_SG_Si40LWZ3Um55cmlpMms

**Comment by m-murad** [comment]
Metrics original - 
metrics-original
Metrics search - 
metrics-search

Fav Triggers original
fav-triggers-original
Fav triggers search
fav-triggers-search

Triggers original
triggers-original
Triggers search
triggers-search

Alerts original
alerts-original
Alerts search
alerts-search

File: mobile/src/main/java/org/hawkular/client/android/adapter/FavMetricsAdapter.java [hunk 1]
```diff
@@ -75,37 +76,40 @@ public int getItemCount() {
...
     public RecyclerViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_list_item_token, parent, false);
+        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+        View view = layoutInflater.inflate(R.layout.layout_list_item_metrics, parent, false);
         view.setTag(new RecyclerViewHolder(view));
         return new RecyclerViewHolder(view);
     }
 
     @Override
-    public void onBindViewHolder(RecyclerViewHolder holder, final int position) {
```
Comment (KeenWarrior) [inline mobile/src/main/java/org/hawkular/client/android/adapter/FavMetricsAdapter.java:84]: Why we should not use `final` directly here?

**Comment by KeenWarrior** [comment]
@free4murad Great work. Nice but big PR. Could break in changing basic UI first and other work in later stages. Will need some time to review this big one

File: mobile/src/main/java/org/hawkular/client/android/adapter/FavMetricsAdapter.java (hunk 1 above)
Comment (m-murad) [reply mobile/src/main/java/org/hawkular/client/android/adapter/FavMetricsAdapter.java:84]: Sure
We can totally do this.
Let me update the PR.

**Comment by m-murad** [comment]
Should I break the PR into two parts
1. UI changes
2. Implementing search functionality

 or is it fine like this?

**Comment by KeenWarrior** [comment]
@free4murad If you break then that will be much better

**Comment by m-murad** [comment]
@garg-anuj 
Should I send multiple PRs
or single PR with multiple commits

**Comment by KeenWarrior** [comment]
@free4murad multiple PRs will be good idea

**Pull request closed without merging**